src/megatron/energon/epathlib/epath.py

```diff
@@ -74,6 +75,24 @@ def __init__(
             else:
                 raise ValueError(f"Unknown protocol: {protocol}")
 
+    @staticmethod
+    def _resolve(path: Union[str, PurePosixPath]) -> PurePosixPath:
+        if isinstance(path, str):
+            path = PurePosixPath(path)
+        parts = path.parts
+        if parts[0] != "/":
+            raise ValueError("Only absolute paths are supported")
+        if ".." in parts or '.' in parts:
+            new_parts = []
+            for part in parts:
+                if part == '..':
+                    new_parts.pop()
+                else:
+                    new_parts.append(part)
+            parts = new_parts
+            path = PurePosixPath("/", *parts)
```

Comment:
Above we are checking that the first part is `/` (absolute path) and here we are adding another `/`?

Reply:
Fixed. Strange that it worked nevertheless